examples/sycl/06_bmg_flash_attention/bmg_flash_attn_prefill_cachedKV_runner.hpp

```diff
@@ -528,27 +542,34 @@ template <class FMHAPrefillCachedKernel, bool isVarLen> struct ExampleRunner {
     block_ref_O.reset(batch * num_heads_q * seq_len_qo * head_size_vo);
 
     if (options.use_paged_kv) {
-        paged_kv_cache.page_size = options.page_size;
-        int num_pages_per_seq = seq_len_kv_cache / paged_kv_cache.page_size;
-        paged_kv_cache.num_pages_per_seq = num_pages_per_seq;
-
-        int num_pages = num_pages_per_seq * batch;
-        paged_kv_cache.page_table.reset(batch * num_pages_per_seq);
-
-        // initialize block table with random mapping for non-contiguous layout
-        std::vector<int> page_mapping(batch * num_pages_per_seq);
-        for (int b = 0; b < batch; ++b) {
-          std::vector<int> physical_pages(num_pages_per_seq);
-          std::iota(physical_pages.begin(), physical_pages.end(), 0);
-          // shuffle physical pages
-          std::shuffle(physical_pages.begin(), physical_pages.end(), std::mt19937{ std::random_device{}() });
-          for (int blk = 0; blk < num_pages_per_seq; ++blk) {
-            int logical_idx = b * num_pages_per_seq + blk;
-            page_mapping[logical_idx] = physical_pages[blk];
-          }
+      paged_kv_cache.page_size = options.page_size;
+      std::vector<int> num_pages_per_seq{0};
+      int num_pages = 0;
+      for(int b = 0; b < cute::get<0>(problem_shape); b++) {
+        int seq_len_cache = isVarLen ? cumulative_seqlen_kv_cache[b + 1] - cumulative_seqlen_kv_cache[b] : seq_len_kv_cache;
+        int pages_per_seq = ceil_div(seq_len_cache, paged_kv_cache.page_size);
+        num_pages_per_seq.push_back(num_pages_per_seq.back() + pages_per_seq);
+        num_pages += pages_per_seq;
+      }
+      paged_kv_cache.page_table.reset(num_pages);
+
+      // initialize block table with random mapping for non-contiguous layout
+      std::vector<int> page_mapping(num_pages);
+      for (int b = 0; b < cute::get<0>(problem_shape); ++b) {
+        std::vector<int> physical_pages(num_pages_per_seq[b + 1] - num_pages_per_seq[b]);
+        std::iota(physical_pages.begin(), physical_pages.end(), 0);
+        // shuffle physical pages
+        std::shuffle(physical_pages.begin(), physical_pages.end(), std::mt19937{ std::random_device{}() });
+        for (int blk = 0; blk < physical_pages.size(); ++blk) {
+          int logical_idx = num_pages_per_seq[b] + blk;
+          page_mapping[logical_idx] = physical_pages[blk];
         }
```

Comment:
Maybe this is correct, but I think the `page_mapping` for *each seq* will contain the same indices (though shuffled differently). 

In other words, the value e.g. `0` will appear `cute::get<0>(problem_shape)` times in the final `page_mapping`.

Is this expected?